airflow/jobs/scheduler_job.py

```diff
@@ -684,23 +685,25 @@ def _execute(self) -> None:
 
             self.register_signals()
 
-            self.processor_agent.start()
+            if not self._standalone_dag_processor and self.processor_agent:
```

Comment:
```suggestion
            if self.processor_agent:
```

same reason as above. `_stamdalone_dag_processor` and `self.processor_agent` are correlated. No need to check "standalone" one - it will never be True when `self.processor_agent` is set.